wire-runtime/src/test/java/com/squareup/wire/protos/redacted/NotRedacted.java

```diff
@@ -50,13 +90,22 @@ public boolean equals(Object other) {
   public int hashCode() {
     int result = hashCode;
     if (result == 0) {
-      result = a != null ? a.hashCode() : 0;
+      result = result * 37 + (a != null ? a.hashCode() : 0);
       result = result * 37 + (b != null ? b.hashCode() : 0);
       hashCode = result;
     }
     return result;
   }
 
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    toStringValue(sb, "a", a);
```

Comment:
If you are generating code anyway, you should generate type-specific code for each field. But I don't think we should generate toString() at all.